### PR TITLE
Simulation frontend integration

### PR DESCRIPTION
**Pull request by Anu-Asok**
Changed sims-list layout
Added debounce search
Custom footer
Changed editable fields to normal fields

### CODE REVIEW

File: src/components/common/P5SimModal.tsx [hunk 1]
```diff
@@ -62,12 +63,51 @@ const P5SimModal = (props: any) => {
...
+
+  if(sim)
+  console.log(`https://editor.p5js.org/${sim.owner}/embed/${sim._id}`);
   return (
     <Modal
       {...props}
-      onOk={handleOKClick}
-      onCancel={handleCancelClick}
       width={width + 50}
+      footer={footerArray(modalType)}
```
Comment (JithinKS97) [inline src/components/common/P5SimModal.tsx]: Is it possible to pass this footer argument as an argument to p5SimModal so as to not pollute this React Component?

File: src/components/common/P5SimModal.tsx [hunk 2]
```diff
@@ -62,12 +63,51 @@ const P5SimModal = (props: any) => {
     bottomRight: true,
   };
 
+  const handleAddToWorkbook = () => {
+    
+    console.log('add sim to workbook');
```
Comment (JithinKS97) [inline src/components/common/P5SimModal.tsx]: Can we just keep the logs that are necessary and remove the rest?

File: src/components/sim-repo/sims-list/SimDetailsEditable.tsx [hunk 3]
```diff
@@ -26,69 +24,49 @@ const SimDetails = (props: Props) => {
...
   };
 
-  const setImgUrl = (imgUrl: string) => {
-    updateSelectedSim(imgUrl, "imgUrl");
+  const setImageURL = (imageURL: string) => {
+    updateSelectedSim(imageURL, "imageURL");
   };
 
-  const Title = () => (
-    <EditableElement
```
Comment (JithinKS97) [inline src/components/sim-repo/sims-list/SimDetailsEditable.tsx:42]: We can delete this component.

**Review by JithinKS97 — COMMENTED** [review]
Keep an eye on the sonar cloud code smell report. Try to minimize the code smells.

File: src/components/sim-repo/sims-list/SimDetailsEditable.tsx (hunk 3 above)
Comment (anandsimmy) [reply src/components/sim-repo/sims-list/SimDetailsEditable.tsx:42]: is our current branch having any build issues?..i can see travis ci keeps failing for both branch and pr

Comment (JithinKS97) [reply src/components/sim-repo/sims-list/SimDetailsEditable.tsx:42]: There is a test that is failing. @Anu-Asok has to take care of it and a couple of changes before we can merge this. The dev branch build is fine.

**Review by JithinKS97 — APPROVED** [review]
@Anu-Asok Thank you !!

**Pull request merged**